File: Source/Tests/ForAlternetStudio/Resources/Debugger.Python/HelloWorld/MyForm.py
import MyForm_Designer
from MyForm_Designer import *

class MyForm (MyForm_Designer): 
    """A form class"""
   
    def __init__(self):
        super().__init__()
        self.InitializeComponent();
  
        
    @staticmethod
    def  RunForm():
        form = MyForm()
        form.ShowDialog()
        form.Dispose()
	
    def CloseButton_Click(self, sender, args):
        self.Close()

    def CloseButton_MouseClick(self, sender, e):
      System.Windows.Forms.MessageBox.Show("click")
    
    def listBox1_DragOver(self, sender, e):
      System.Windows.Forms.MessageBox.Show("Drag over")

System.Console.WriteLine("Hello from Python")
MyForm.RunForm()

  

File: Source/Tests/ForAlternetStudio/Resources/Designer.TypeScript/WinForms-dotnetcore/JS/MyForm.js
 
///<reference path="clr.d.ts" />

class MyForm extends MyForm_Designer
{ 

   
    constructor()   
	{ 
		   
		super(); 
		super.InitializeComponent(); 
	}
     
    Dispose() 
	{
		System.Console.WriteLine("Dispose");
		this.clrSuper.Dispose();
	}

    static RunForm()
    {
        var form = new MyForm();
        form.ShowDialog();
        form.Dispose();
    }
	
    CloseButton_Click(sender, args) 
    {
        this.Close(); // Close is not recognized, while this.Close works fine.
    }

    CloseButton_MouseClick(sender, e)
    {
		System.Windows.Forms.MessageBox.Show("click");
    }

    listBox1_DragOver(sender, e)
    {
		
    }

    listBox1_SelectedIndexChanged(sender, e)
    {
        
    }
}


System.Console.WriteLine("Hello from JS 1");

MyForm.RunForm();
System.Console.WriteLine("Hello from JS 2");

  
